flink-table/flink-table-planner-blink/src/main/java/org/apache/calcite/rel/core/Join.java

```diff
@@ -0,0 +1,341 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.rel.core;
```

Comment:
Maybe we should try to avoid making this package exactly the same with Calcite's? It will make developer confusing which one did we import.

Reply:
If packages are different betweent Flink `Join` and Calcite `Join`, we need to copy all codes associated with `Join` from Calcite to Flink. This class will be delete when calcite-1.20 is released. (may be one month later)